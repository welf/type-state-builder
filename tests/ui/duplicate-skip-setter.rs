use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct DuplicateSkipSetter {
    name: String,

    // This should be an error: duplicate skip_setter attributes
    #[builder(skip_setter, default = "42")]
    #[builder(skip_setter)]
    count: u32,
}

fn main() {
    let _instance = DuplicateSkipSetter::builder()
        .name("test".to_string())
        .build();
}