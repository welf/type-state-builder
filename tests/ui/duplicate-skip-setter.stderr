error: Duplicate skip_setter attribute. Only one skip_setter is allowed per field
 --> tests/ui/duplicate-skip-setter.rs:9:15
  |
9 |     #[builder(skip_setter)]
  |               ^^^^^^^^^^^
